star-navigation/src/components/bottom-container/time-slider.tsx

```diff
@@ -13,21 +15,29 @@ interface IProps {
   disabled?: boolean;
 }
 
-export const TimeSlider: React.FC<IProps> = ({ value, month, onChange, disabled }) => {
+export const TimeSlider: React.FC<IProps> = ({ value, day, month, onChange, disabled }) => {
+  const { observerLat, observerLong } = config;
+
   const handleChange = (e: Event, val: number | number[]) => {
     onChange(e, typeof val === "number" ? val : val[0]);
   };
 
-  const { sunrise, sunset } = sunriseSunset[month - 1];
-
-  const sunriseWidth = `${100 * sunrise / 24}%`;
-  const sunsetStart = `${100 * sunset / 24}%`;
-  const sunsetWidth = `${100 * (24 - sunset) / 24}%`;
+  const date = new Date(getDateTimeString({ month, day, timeOfDay: value }));
+  const times = getTimes(date, observerLat, observerLong);
+  const sunrise = dateToFractionalHoursInRightTimezone(times.sunrise);
+  const sunset = dateToFractionalHoursInRightTimezone(times.sunset);
 
   const Rail = () => (
     <div className={css.customRail}>
-      <div className={css.beforeSunrise} style={{ width: sunriseWidth }} />
-      <div className={css.afterSunset} style={{ left: sunsetStart, width: sunsetWidth }} />
+      {
+        sunrise < sunset ?
```

Comment:
Isn't `sunrise` always before `sunset`? Is this to handle oddball situations like being at a pole?

Reply:
In terms of universe and absolute time, it's true. :) But this handles situations when sunset is after midnight, so in terms of hour values (0-24), it's smaller than sunrise. For example, the sunset value can be 00:30 am, and sunrise at 6 am. It doesn't need to be a pole, Hooper Bay is north enough for that to happen.